### ISSUE TITLE
allow list for authentication

### ISSUE THREAD

**Issue by bedis** [haproxytech/kubernetes-ingress#349]
Current auth feature is configured in the frontend section of HAProxy.
This is fine for most cases.
That said, some people may want to allow unauthenticated browsing for some conditions (mainly source IP range for example), and doing this per ingress rule.

This could be achieved by a config-snippet in the backend, coupled to a global config snippet in the configmap:

* configmap:
```
controller:
  config:
    global-config-snippet: |
      userlist default-mylist
        group authenticated-users users foo
        user foo password ******* groups authenticated-users
```

* ingress rule config snippet:
```
    haproxy.org/backend-config-snippet: |
      acl auth_ok http_auth_group(default-mylist) authenticated-users
      acl my_subnets src 10.0.0.0/8
      http-request auth unless auth_ok || my_subnets
```

that said, the configmap part is usually not accessible to developper when they deploy their ingress rules.

To summarize, I need 2 things:
- a way to set userlist per ingress rule
- a way to set a bypass condition for the authentication rule

**Comment by Mo3m3n** [comment]
>To summarize, I need 2 things:
>
>   a way to set userlist per ingress rule
>   a way to set a bypass condition for the authentication rule

The [basic-auth annotations](https://github.com/haproxytech/kubernetes-ingress/tree/master/documentation#authentication) are available per ingress, so this should answer the first requirement.

I have added to the "enhancement" label to work on an annotation which value can be used as a haproxy acl to bypass authentication.

**Comment by renchap** [comment]
Is this still in your ideas list?

I have a use case for this feature where I need to ignore authentication for a set of paths that are used as callbacks for some asynchronous processes, but I have not found a way to do it with the ingress
